src/puppetlabs/kitchensink/classpath.clj

```diff
@@ -35,6 +35,15 @@
   [cl]
   (dp/addable-classpath? cl))
 
+(defn- ensure-modifiable-classloader
+  "Check if there is a modifiable classloader in the current hierarchy, and add
+  one if not."
+  []
+  (let [classloader (.. Thread currentThread getContextClassLoader)]
+    (when-not (last (filter modifiable-classloader? (classloader-hierarchy classloader)))
+      (let [new-cl (clojure.lang.DynamicClassLoader. classloader)]
```

Comment:
Is it okay to just use the clojure one for this? We could implement our own pretty easily but it seemed to have what was needed.

Reply:
this seems fine?

Reply:
https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/DynamicClassLoader.java#L33-L38 isn't super readable, but it seems fine and hasn't been touched in years.